clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -116,7 +118,8 @@ public KubernetesV2Credentials(
 
     this.kubectlExecutable = managedAccount.getKubectlExecutable();
     this.kubectlRequestTimeoutSeconds = managedAccount.getKubectlRequestTimeoutSeconds();
-    this.kubeconfigFile = managedAccount.getKubeconfigFile();
+    this.kubeconfigFile =
+        configFileService.getLocalPath(managedAccount.getKubeconfigFile(), "kube", "config");
```

Comment:
Should this be just calling `getKubeconfigFile()` directly (which will itself either fetch the kube config or write the contents to a file)?

Reply:
Do you mean `KubernetesNamedAccountCredentials.CredentialFactory#getKubeconfigFile`? It would make sense for `KubernetesNamedAccountCredentials.CredentialFactory#buildV2Credentials` to pass the results of `getKubeconfigFile` into the `KubernetesV2Credentials` ctor, since the defaulting of the file location would also happen then.

Reply:
Yes, that is what I meant, though looking at it more closely (assuming I'm reading it correctly) it looks like the V2 provider always just directly read `kubeconfigFile` instead of going through the logic to either read `kubeconfigFile` or write `kubeconfigContents` to a file so this change correctly maintains that behavior.  (I suspect that difference between V1 and V2 is not deliberate, but that's not something we need to investigate here.)

If we do want to make V1 and V2 consistent, I agree that consolidating the logic to live in `KubernetesNamedAccountCredentials` and passing the result of `getKubeconfigFile` to both constructors makes sense.  But I think upon a closer read now that what you have is fine given that it replicates the existing behavior.

Reply:
It's definitely hard to figure out what's intentional and not intentional when looking at those differences without any historical context :-). I've made the recommended change and it's pretty clean, so I'm fine committing that or leaving as-is.

Reply:
I've you've made that change, I'd say go for it...at some point I plan to further refactor the V1/V2 to share more code so that's a step towards that (where they at least treat getting the kube config the same way).

Reply:
Resolved in 2nd commit.